traverson4j-hc5/src/main/java/uk/co/autotrader/traverson/http/ApacheHttpTraversonClientAdapter.java

```diff
@@ -37,32 +37,42 @@ public ApacheHttpTraversonClientAdapter(CloseableHttpClient client) {
 
     @Override
     public <T> Response<T> execute(Request request, Class<T> returnType) {
-        HttpUriRequest httpUriRequest = apacheHttpUriConverter.toRequest(request);
+        ClassicHttpRequest httpRequest = apacheHttpUriConverter.toRequest(request);
 
         HttpClientContext clientContext = HttpClientContext.create();
-        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
         AuthCache authCache = new BasicAuthCache();
 
         for (AuthCredential authCredential : request.getAuthCredentials()) {
-            UsernamePasswordCredentials userPassword = new UsernamePasswordCredentials(authCredential.getUsername(), authCredential.getPassword());
-            AuthScope scope = AuthScope.ANY;
+            UsernamePasswordCredentials userPassword = new UsernamePasswordCredentials(authCredential.getUsername(), authCredential.getPassword().toCharArray());
+            AuthScope authScope = new AuthScope(null, null, -1, null, null);
```

Comment:
Worth extracting this as a constant describing what it means, given nulls and -1 feel bit like magic?

Reply:
yeah will do